src/non_markov_localization.h

```diff
@@ -47,6 +47,11 @@ struct VisibilityGlobConstraint;
 
 class NonMarkovLocalization {
  public:
+
+  // Typedef for timestamps. First entry is the seconds, second entry is
+  // nanoseconds.
+  typedef std::pair<uint32_t, uint32_t> Timestamp;
```

Comment:
The problem with an opaque unnamed-pair is that it's easy to forget and misuse which part is the seconds and which the nanoseconds. 
You can use the standard `timespec` instead: https://en.cppreference.com/w/c/chrono/timespec